### PR TITLE
Fix 2ndheader encoding

### PR DESCRIPTION
**Pull request by michal-josef-spacek**

### CODE REVIEW

**Comment by rurban** [comment]
Yes, it fixed a few json-check oda errors.

**Comment by rurban** [comment]
We should store the SECTION_2NDHEADER_R13_SIZE part in some junk field

**Comment by michal-josef-spacek** [comment]
I added info about structure to log.

**Review by rurban — CHANGES_REQUESTED** [review]

File: include/dwg.h [hunk 1]
```diff
@@ -9041,6 +9041,11 @@ typedef enum DWG_SECTION_TYPE_R13
   SECTION_THUMBNAIL_R13 = 6,
 } Dwg_Section_Type_r13;
 
+typedef enum DWG_SECTION_SIZE_R13
```
Comment (rurban) [inline include/dwg.h:9044]: I'd prefer this as define, not enum.

File: include/dwg.h [hunk 2]
```diff
@@ -9633,6 +9638,7 @@ typedef struct _dwg_secondheader
   Dwg_SecondHeader_Handles handles[14];
   BITCODE_RS crc;
   BITCODE_RLL junk_r14; /*!< r14-r2000 */
+  BITCODE_RC before_sentinel[53];
```
Comment (rurban) [inline include/dwg.h:9641]: and use the define here

**Review by rurban — CHANGES_REQUESTED** [review]

File: src/encode.c [hunk 3]
```diff
@@ -3607,23 +3609,7 @@ dwg_encode (Dwg_Data *restrict dwg, Bit_Chain *restrict dat)
...
-          dwg->header.section[4].size = UINT64_C (4);
-          _obj->sections[4].size = UINT32_C (4);
-          dat->byte = pvzadr;
-          encode_secondheader_private (dat, dwg);
-        }
-
-      _obj->size = encode_patch_RLsize (dat, pvzadr);
-      if (_obj->size < dat->byte)
-        dwg->header.section[SECTION_2NDHEADER_R13].size = _obj->size;
+      encode_patch_RLsize (dat, pvzadr);
```
[the remark this answers is not in the document]
Comment (michal-josef-spacek) [reply src/encode.c:3611]: My code is totally wrong

File: include/dwg.h (hunk 1 above)
Comment (michal-josef-spacek) [reply include/dwg.h:9044]: My code is totally wrong

File: include/dwg.h (hunk 2 above)
Comment (michal-josef-spacek) [reply include/dwg.h:9641]: My code is totally wrong

**Review by rurban — APPROVED** [review]

File: src/encode.c [hunk 4]
```diff
@@ -3527,10 +3518,6 @@ dwg_encode (Dwg_Data *restrict dwg, Bit_Chain *restrict dat)
       BITCODE_BL vcount;
 
       assert (dat->byte);
```
Comment (rurban) [inline src/encode.c:3520]: add encode_objfreespace here

File: src/decode.c [hunk 5]
```diff
@@ -916,7 +918,27 @@ decode_R13_R2000 (Bit_Chain *restrict dat, Dwg_Data *restrict dwg)
    */
 
   /*-------------------------------------------------------------------------
-   * Second header, section 3. R13c3-R2000 only.
+   * AcDb:ObjFreeSpace, section 3
+   */
+
+  if (dwg->header.sections > 3
+      && dwg->header.section[SECTION_OBJFREESPACE_R13].address != 0)
```
Comment (rurban) [inline src/decode.c:925]: address is usually 0. rather search for the sentinel, and check if object_end equals dat->byte - 53. if there's room for this section before 2ndheader

**Review by rurban — CHANGES_REQUESTED** [review]
see my work/r13-objfreespace

**Comment by rurban** [comment]
Merged with fixes

**Pull request closed without merging**